### PR TITLE
Remove STLTuple

### PR DESCRIPTION
**Pull request by j-stephan**
The SYCL back-end and the late OpenMP 5 / OpenACC back-ends made use of Codeplay's tuple implementation to pass parameter packs into kernels. STLTuple is no longer maintained, the repository has been archived. Tests on my local machine have shown that it is now possible to use `std::tuple` instead. This PR replaces all usages of STLTuple with `std::tuple`.

### CODE REVIEW

**Review by bernhardmgruber — APPROVED** [review]

**Comment by bernhardmgruber** [comment]
Those are my favorite PRs!

File: include/alpaka/kernel/TaskKernelGenericSycl.hpp [hunk 1]
```diff
@@ -44,7 +42,7 @@
                 sycl::nd_item<TDim::value> work_item) [[intel::reqd_sub_group_size(sub_group_size)]]                  \
             {                                                                                                         \
                 auto acc = TAcc{item_elements, work_item, dyn_shared_accessor, st_shared_accessor};                   \
-                core::apply(                                                                                          \
+                std::apply(                                                                                           \
```
Comment (j-stephan) [inline include/alpaka/kernel/TaskKernelGenericSycl.hpp:45]: Side note: These white spaces behind the opening `(` are added by clang-format. Removing them makes the CI job complain.

Comment (bernhardmgruber) [reply include/alpaka/kernel/TaskKernelGenericSycl.hpp:45]: Well, they are necessary to align the `\` at the end of the line.

**Comment by psychocoderHPC** [comment]
Ohh we are currently building code based on alpaka's Tuple implementation because we would like to pass classes with tuple inside onto the GPU what is not possible with `std::tupel` because it is not trivially copyable.

**Comment by j-stephan** [comment]
I see. I'll convert this to a draft so we can discuss on how to proceed.

**Comment by psychocoderHPC** [comment]
We can always pull the tuple implementation into PIConGPU but avoid this because alpaka is shipping it anyway.

**Comment by bernhardmgruber** [comment]
I would argue that this dependence could move downstream into picongpu, so removing it from alpaka is fair.

**Comment by j-stephan** [comment]
I agree with Bernhard here. STLTuple wasn't intended to be exposed as a public interface - user code depending on it should pull it in on its own.

**Comment by j-stephan** [comment]
@psychocoderHPC Would it be okay for you to pull in STLTuple directly into the affected code bases?

**Review by psychocoderHPC — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
offline discussed: we decided in the developer meeting to remove the code because we do not provide CI tests.
We discussed introducing an own implementation in the future but this requires that we test the implementation.

**Pull request merged**